sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -251,8 +251,8 @@ def tune(
             pip_index_url: The PyPI url from which to install Python packages.
             metrics_collector_config: Specify the config of metrics collector, 
                 for example, `metrics_collector_config = {"kind": "Push"}`.
-                Currently, we only support `StdOut` and `Push` metrics collector.
-
+                for using custom metric collectors use "custom_collector" key,
+                for example, `metrics_collector_config = {"custom_collector": "prometheus"}`.
```

Comment:
I guess `prometheus` is not a custom collector, since we define it here:

https://github.com/kubeflow/katib/blob/8eb0e86385a5b54d67fd775ff930f4f038b5b370/pkg/apis/controller/common/v1beta1/common_types.go#L216-L220

Here are some relevant resources which may be helpful for you:
https://github.com/kubeflow/katib/blob/master/examples/v1beta1/metrics-collector/custom-metrics-collector.yaml

Reply:
Btw, `custom_collector` accepts `V1Container` as its input, not a `str`:

https://github.com/kubeflow/katib/blob/8eb0e86385a5b54d67fd775ff930f4f038b5b370/sdk/python/v1beta1/kubeflow/katib/models/v1beta1_collector_spec.py#L21-L43

Can you add some comments to remind users of this usage?

Reply:
Also, can you add some e2e tests like this (an example for `StdOut` Collector)?

https://github.com/kubeflow/katib/blob/8eb0e86385a5b54d67fd775ff930f4f038b5b370/test/e2e/v1beta1/scripts/gh-actions/run-e2e-tune-api.py#L16

Reply:
Thanks @Electronic-Waste, I have made the appropriate changes in the function comments to reflect the correct usage

for e2e tests, should I modify the existing test to use `metrics_collector_config` or should I create a separate test script for it altogether

Reply:
@prakhar479 I think modifying the existing `run-e2e-tune-api.py` test is better since it's hard to pass the collector config to the test script: https://github.com/kubeflow/katib/blob/master/test/e2e/v1beta1/scripts/gh-actions/run-e2e-tune-api.sh. WDYT👀 @andreyvelich @tenzen-y @johnugeorge 

Also a small tip for reference: you can build images you need here

https://github.com/kubeflow/katib/blob/8eb0e86385a5b54d67fd775ff930f4f038b5b370/test/e2e/v1beta1/scripts/gh-actions/build-load.sh#L166-L170

Reply:
Thanks @Electronic-Waste for the helpful insights 😄